In [1]:
from pathlib import Path
import pandas as pd
from sklearn.model_selection import train_test_split

PROJECT_ROOT = Path.cwd().parent

X_full = pd.read_csv(
    PROJECT_ROOT / "data" / "processed" / "student_full_information.csv"
)

X_early = pd.read_csv(
    PROJECT_ROOT / "data" / "processed" / "student_early_warning.csv"
)

# Load target from encoded dataset
df_encoded = pd.read_csv(
    PROJECT_ROOT / "data" / "processed" / "student_data_encoded.csv"
)

y = df_encoded["G3"]

print("X_full:", X_full.shape)
print("X_early:", X_early.shape)
print("y:", y.shape)

X_full: (395, 41)
X_early: (395, 39)
y: (395,)


In [2]:
Xtr_f, Xte_f, ytr, yte = train_test_split(
    X_full,
    y,
    test_size=0.20,
    random_state=42
)

Xtr_e, Xte_e, _, _ = train_test_split(
    X_early,
    y,
    test_size=0.20,
    random_state=42
)

print("Train/Test split complete.")

Train/Test split complete.


In [3]:
print("Full-information training:", Xtr_f.shape)
print("Full-information testing :", Xte_f.shape)

print("Early-warning training:", Xtr_e.shape)
print("Early-warning testing :", Xte_e.shape)

print("Training target:", ytr.shape)
print("Testing target :", yte.shape)

Full-information training: (316, 41)
Full-information testing : (79, 41)
Early-warning training: (316, 39)
Early-warning testing : (79, 39)
Training target: (316,)
Testing target : (79,)


In [4]:
print(
    "Identical training indices:",
    Xtr_f.index.equals(Xtr_e.index)
)

print(
    "Identical testing indices:",
    Xte_f.index.equals(Xte_e.index)
)

Identical training indices: True
Identical testing indices: True


In [5]:
output_dir = PROJECT_ROOT / "data" / "processed"

Xtr_f.to_csv(output_dir / "X_train_full.csv", index=False)
Xte_f.to_csv(output_dir / "X_test_full.csv", index=False)

Xtr_e.to_csv(output_dir / "X_train_early.csv", index=False)
Xte_e.to_csv(output_dir / "X_test_early.csv", index=False)

ytr.to_csv(output_dir / "y_train.csv", index=False)
yte.to_csv(output_dir / "y_test.csv", index=False)

print("All train/test files saved.")

All train/test files saved.


# Session 22: Train/Test Split

## Objective

Create reproducible train/test splits for both modeling scenarios.

## Results

- Full-information training set: (316, 41)
- Full-information test set: (79, 41)

- Early-warning training set: (316, 39)
- Early-warning test set: (79, 39)

## Interpretation

An 80/20 split was used to create separate training and testing datasets.

The same random seed (42) was used for both scenarios so that the exact same students appear in the training and testing sets. This ensures a fair comparison between the full-information and early-warning models.

The training datasets will be used for model fitting, while the testing datasets will be reserved for unbiased evaluation of predictive performance.